### PR TITLE
feat: openapi parser array, null, enum, oneof, allof

### PR DESCRIPTION
**Pull request by RohinBhargava**
## Short description of the changes made

* Changed structure of nodes slightly to accommodate better ergonomics -- convert creates more natural fdr shapes, now
* Added Array, Null, Enum, and OneOf nodes
* Added allOf support for object nodes
* Added Deeptune snapshot test
* Would not expect this to be at full parity with the openapi parser, but is continuing to improve

## What was the motivation & context behind this PR?

* Fleshing out the OpenApi Schema Parser
* Will work on endpoint parser, availability and displayName extensions next

## How has this PR been tested?

* Unit tests added for each new node,
* Added Deeptune snapshot test

### CODE REVIEW

File: packages/ui/fern-dashboard/src/routeTree.gen.ts [hunk 1]
```diff
@@ -88,16 +91,81 @@ declare module '@tanstack/react-router' {
...
 // Create and export the route tree
 
-export const routeTree = rootRoute.addChildren({
-  IndexRoute,
-  AuthRoute: AuthRoute.addChildren({ AuthLoginRoute }),
-  AuthenticatedRoute: AuthenticatedRoute.addChildren({
-    AuthenticatedTeamOrgIdRoute,
-  }),
-})
+interface AuthRouteChildren {
```
Comment (dsinghvi) [inline packages/ui/fern-dashboard/src/routeTree.gen.ts:94]: nit: lets remove this? or is this normal?

Comment (RohinBhargava) [reply packages/ui/fern-dashboard/src/routeTree.gen.ts:94]: oh this is weird, I think this got added on compile, thanks for catching

**Review by abvthecity — APPROVED** [review]

File: packages/parsers/src/openapi/3.1/schemas/ArrayConverter.node.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export class ArrayConverterNode extends BaseOpenApiV3_1Node<
+    ArrayConverterNode.Input,
+    ArrayConverterNode.Output | undefined
+> {
+    innerSchema: SchemaConverterNode | undefined;
+
+    constructor(...args: BaseOpenApiV3_1NodeConstructorArgs<ArrayConverterNode.Input>) {
+        super(...args);
+        this.safeParse();
```
Comment (dsinghvi) [inline packages/parsers/src/openapi/3.1/schemas/ArrayConverter.node.ts:24]: curious -- why do we call `this.safeParse()` in the constructor? if we do this everywhere can we push this to `BaseOpenApiV3_1Node` ?

File: packages/parsers/src/openapi/3.1/schemas/BooleanConverter.node.ts [hunk 3]
```diff
@@ -1,41 +1,48 @@
...
+    constructor(...args: BaseOpenApiV3_1NodeConstructorArgs<BooleanConverterNode.Input>) {
+        super(...args);
+        this.safeParse();
+    }
 
-        if (input.default != null && typeof input.default !== "boolean") {
+    parse(): void {
+        if (this.input.default != null && typeof this.input.default !== "boolean") {
             this.context.errors.warning({
                 message: "The default value for a boolean type should be a boolean",
```
Comment (dsinghvi) [inline packages/parsers/src/openapi/3.1/schemas/BooleanConverter.node.ts:29]: nit (unrelated to this PR) might be nice for the error message here `Expected example to be boolean. Received {}`

Comment (RohinBhargava) [reply packages/parsers/src/openapi/3.1/schemas/BooleanConverter.node.ts:29]: This is a good point, will fix all the error messages to accommodate this messaging.

File: packages/parsers/src/openapi/3.1/schemas/ArrayConverter.node.ts (hunk 2 above)
Comment (RohinBhargava) [reply packages/parsers/src/openapi/3.1/schemas/ArrayConverter.node.ts:24]: This is an unfortunate part of TS. The issue is if there is a superclass call, the `this` doesn't get bound properly, and I believe the object members get associated with the proto of the super class (not 100% the cause, but 100% sure on the symptoms). The only real way to do this is to unfortunately include the constructor, and this is the slimmest I can make it.

File: packages/parsers/src/openapi/3.1/schemas/ArrayConverter.node.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    parse(): void {
+        this.innerSchema = new SchemaConverterNode(this.input.items, this.context, this.accessPath, "items");
+
+        if (this.input.items == null) {
+            this.context.errors.error({
+                message: "No items found in array",
```
Comment (dsinghvi) [inline packages/parsers/src/openapi/3.1/schemas/ArrayConverter.node.ts:32]: nit: `item` key missing in array schema

File: packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts [hunk 5]
```diff
@@ -20,114 +24,109 @@ export class SchemaConverterNode extends BaseOpenApiV3_1Node<
...
-        pathId: string,
-    ) {
-        super(input, context, accessPath, pathId);
+    parse(): void {
+        // Check if the input is a reference object
+        if (isReferenceObject(this.input)) {
+            this.typeShapeNode = new ReferenceConverterNode(this.input, this.context, this.accessPath, this.pathId);
+        } else {
+            // If the object is not a reference object, then it is a schema object, gather all appropriate variables
+            this.name = this.input.title;
```
Comment (dsinghvi) [inline packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts:43]: what is the `name` used for?

File: packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts [hunk 6]
```diff
@@ -20,114 +24,109 @@ export class SchemaConverterNode extends BaseOpenApiV3_1Node<
...
+                        case "null":
+                            this.typeShapeNode = new NullConverterNode(
+                                this.input as NullConverterNode.Input,
+                                this.context,
+                                this.accessPath,
+                                this.pathId,
+                            );
+                            break;
+                        default:
+                            new UnreachableCaseError(this.input.type);
```
Comment (dsinghvi) [inline packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts:115]: what does `UnreachableCaseError` do?

Comment (RohinBhargava) [reply packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts:115]: UnreachableCaseError is effectively the same as assertNever; it is used in more places in the platform packages, so decided to go with that utility

File: packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts (hunk 5 above)
Comment (RohinBhargava) [reply packages/parsers/src/openapi/3.1/schemas/SchemaConverter.node.ts:43]: it's used in the Components -> Types conversion, and from OneOf schemas (for display names)

**Pull request merged**